File: 50/source.java
public class e50 {
	
	static boolean[] primes  = new boolean[100000001];
	static final int NUM = 1000000;
	
	private static void fillPrimes(int num){
		for(int i=0; i < num;i++){
		primes[i] = (!(i == 2 || i%2 == 0 || i == 0 || i == 1));
		}
		for(int i = 3; i <= (num/2); i = i + 2){
			if(primes[i]){
				for(int j= 2*i; j < num; j += i){
					primes[j] = false;
					}
				}
			}
		primes[2] = true;
	}
	
	private static int chain(int a){
		int length = 0;
		int sum = 0;
		int max = 0;
		for(int start = 2; start <= Math.sqrt(a)+1; start++){
			length = 0;
			sum = 0;
			/* hopefully makes it faster */
			if(start*length > a){
				break;
			}
			for(int i = start; i <= (a/2)+1; i++){
				if(primes[i]){
					if(sum + i > a){
						break;
					}
					sum += i;
					length++;
					if(length > max && sum == a){
						max = length;
					}
				}
			}
		}
		return max;
	}
	
	public static void main(String[] args) {
		fillPrimes(NUM);
		int prime, max, tmp;
		prime = 0;
		max   = 0;
		/* probably inside this range */
		for(int i = 600000; i < 1000000; i++){
			if(primes[i]){
				tmp = chain(i);
				if(tmp > max){
					max = tmp;
					prime = i;
				}
			}
		}
		System.out.println("Result: " + prime);
	}

}

File: 27/source.py
from math import sqrt

def erathostenos(num):
    primes = { i:(not(i == 2 or i % 2 == 0 or i == 0 or i == 1)) for i in range(num+1) }

    primes[2] = True

    limit   = sqrt(num) + 1
    i       = 3
    while i <= limit:
        if primes[i]:
            j = 2 * i
            while j < num:
                primes[j] = False
                j += i
        i += 2
    return primes

def longestChain(a, b):
    long = 0
    n = 0
    while True:
        val = n**2 + a*n + b
        if val > 0:
            if not primes[val]:
                return (long, a, b)
            long += 1
        n += 1

primes = erathostenos(100000)
maxV = []
for i in range(1000):
    for k in range(1000):
        if primes[i] and primes[k]:
            maxV.append(longestChain(i, k))
            maxV.append(longestChain(-1*i, k))
            maxV.append(longestChain(-1*i, -1*k))
            maxV.append(longestChain(i, -1*k))

maxT = max(maxV)
print( maxT, (maxT[1] * maxT[2]))